### ISSUE TITLE
No-unused-locals does not warn for unused type parameter in declaration file

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#16357]
**TypeScript Version:**  nightly (2.4.0-dev.20170608)

**Code**

```ts
export function f<T>(): void;
```

**Expected behavior:**

Unused type parameter `T`.

**Actual behavior:**

No error.
